src/coreComponents/python/modules/geosx_xml_tools_package/geosx_xml_tools/tests/test_manager.py

```diff
@@ -12,295 +13,309 @@
 
 # Test the unit manager definitions
 class TestUnitManager(unittest.TestCase):
-  @classmethod
-  def setUpClass(cls):
-    cls.tol = 1e-6
+    @classmethod
+    def setUpClass(cls):
+        cls.tol = 1e-6
 
-  def test_unit_dict(self):
-    unitManager.buildUnits()
-    self.assertTrue(bool(unitManager.units))
+    def test_unit_dict(self):
+        unitManager.buildUnits()
+        self.assertTrue(bool(unitManager.units))
 
-  def checkUnits(self, unitStruct, expectedValue):
-    val = float(unitManager(unitStruct))
-    self.assertTrue(abs(val - expectedValue) < self.tol)
+    def checkUnits(self, unitStruct, expectedValue):
+        val = float(unitManager(unitStruct))
+        self.assertTrue(abs(val - expectedValue) < self.tol)
 
-  # Scale value tests
-  def test_scale_a(self):
-    self.checkUnits(['2', 'meter'], 2.0)
+    # Scale value tests
+    def test_scale_a(self):
+        self.checkUnits(['2', 'meter'], 2.0)
 
-  def test_scale_b(self):
-    self.checkUnits(['1.234', 'meter'], 1.234)
+    def test_scale_b(self):
+        self.checkUnits(['1.234', 'meter'], 1.234)
 
-  def test_scale_c(self):
-    self.checkUnits(['1.234e1', 'meter'], 12.34)
+    def test_scale_c(self):
+        self.checkUnits(['1.234e1', 'meter'], 12.34)
 
-  def test_scale_d(self):
-    self.checkUnits(['1.234E1', 'meter'], 12.34)
+    def test_scale_d(self):
```

Comment:
I do not know which testing fwk you are using, but there probably is a parametrization on data feature that you could use. Like https://docs.pytest.org/en/6.2.x/example/parametrize.html e.g.

Reply:
Excellent idea - since I was already using unittests, I've used the parameterized package to greatly decrease the redundant code in the tests.